syft/frameworks/torch/mpc/falcon.py

```diff
@@ -0,0 +1,82 @@
+import torch
+
+from syft.frameworks.torch.tensors.interpreters.replicated_shared import ReplicatedSharingTensor
+from syft.workers.base import BaseWorker
+
+from typing import List
+
+
+class Falcon(object):
+    @staticmethod
+    def _share_random_tensor(
+        field: int, workers: List[BaseWorker]
+    ) -> (ReplicatedSharingTensor, ReplicatedSharingTensor):
+        """
+        TODO: Need to move this in Primitive Store:
+        Share a random tensor with the owners of some shares
+        """
+        c = torch.randint(2, size=(1,))
+        c_share_L = c.share(*workers, protocol="falcon", field=field)
+        c_share_2 = c.share(*workers, protocol="falcon", field=2)
+        return c_share_2, c_share_L
+
+    @staticmethod
+    def select_share(
+        b_share: ReplicatedSharingTensor,
+        x_share: ReplicatedSharingTensor,
+        y_share: ReplicatedSharingTensor,
+    ) -> ReplicatedSharingTensor:
+        """ Performs select share protocol
+        If the bit b_share is 0, x_sh is returned
+        If the bit b_share is 1, y_sh is returned
+
+        Args:
```

Comment:
as a general comment: The sharing used in ReplicatedSharingTensor (https://github.com/OpenMined/PySyft/pull/3856/files) is for semi-honest adversary and falcon is secure against malicious adversary...

Reply:
Yep - currently the RST and the Falcon protocol are "combined" inside the RST (the \pi_mul operation is encapsulated in the RST) - if we add more protocols over RST we can have something like we have in the ```aditive_sharing_tensor.py```
```
@crypto_protocl("fss")
def add(self, other):
    # here we define how our operation over the AST should work if we use the FSS protocol
```